document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementUploadService.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.hmcts.document.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.hateoas.hal.HalLinkDiscoverer;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.retry.annotation.CircuitBreaker;
+import org.springframework.retry.annotation.Recover;
+import org.springframework.stereotype.Service;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.document.model.FileUploadResponse;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Arrays.asList;
+import static java.util.stream.StreamSupport.stream;
+import static uk.gov.hmcts.document.service.UploadRequestBuilder.prepareRequest;
+
+@Service
+public class DocumentManagementUploadService {
+
+    private static final Logger log = LoggerFactory.getLogger(DocumentManagementUploadService.class);
+
+    @Value("${document.management.upload.file.url}")
+    private String documentManagementServiceURL;
+
+    @Autowired
+    private RestTemplate template;
+
+
+    @CircuitBreaker(exclude = HttpClientErrorException.class)
+    public List<FileUploadResponse> uploadFiles(List<MultipartFile> files, String authorizationToken, String requestId) {
+        MultiValueMap<String, Object> parameters = prepareRequest(files);
+
+        HttpHeaders httpHeaders = setHttpHeaders(authorizationToken);
+
+        HttpEntity<MultiValueMap<String, Object>> httpEntity = new HttpEntity<>(parameters,
+            httpHeaders);
+
+        JsonNode filesJsonArray = template.postForObject(documentManagementServiceURL, httpEntity, ObjectNode.class)
```

Comment:
we should be using document management api here not manually doing the http like in IdamApi and the BankHolidaysApi